### ISSUE TITLE
Error on first time *.java file open

### ISSUE THREAD

**Issue by krossovochkin** [hsanson/vim-android#24]
Hi

I downloaded sample android project: https://github.com/chrisbanes/cheesesquare
Installed Vim for Windows (ftp://ftp.vim.org/pub/vim/pc/gvim74.exe)

This is my `_vimrc` file:
```
set nocompatible              " be iMproved, required
filetype off                  " required

" set the runtime path to include Vundle and initialize
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
" alternatively, pass a path where Vundle should install plugins
"call vundle#begin('~/some/path/here')

" let Vundle manage Vundle, required
Plugin 'VundleVim/Vundle.vim'
Bundle 'hsanson/vim-android'

" All of your Plugins must be added before the following line
call vundle#end()            " required
filetype plugin indent on    " required
" To ignore plugin indent changes, instead use:
"filetype plugin on
"
" Brief help
" :PluginList       - lists configured plugins
" :PluginInstall    - installs plugins; append `!` to update or just :PluginUpdate
" :PluginSearch foo - searches for foo; append `!` to refresh local cache
" :PluginClean      - confirms removal of unused plugins; append `!` to auto-approve removal
"
" see :h vundle for more details or wiki for FAQ
" Put your non-Plugin stuff after this line

set nocompatible
source $VIMRUNTIME/vimrc_example.vim
source $VIMRUNTIME/mswin.vim
behave mswin

set diffexpr=MyDiff()
function MyDiff()
  let opt = '-a --binary '
  if &diffopt =~ 'icase' | let opt = opt . '-i ' | endif
  if &diffopt =~ 'iwhite' | let opt = opt . '-b ' | endif
  let arg1 = v:fname_in
  if arg1 =~ ' ' | let arg1 = '"' . arg1 . '"' | endif
  let arg2 = v:fname_new
  if arg2 =~ ' ' | let arg2 = '"' . arg2 . '"' | endif
  let arg3 = v:fname_out
  if arg3 =~ ' ' | let arg3 = '"' . arg3 . '"' | endif
  let eq = ''
  if $VIMRUNTIME =~ ' '
    if &sh =~ '\<cmd'
      let cmd = '""' . $VIMRUNTIME . '\diff"'
      let eq = '"'
    else
      let cmd = substitute($VIMRUNTIME, ' ', '" ', '') . '\diff"'
    endif
  else
    let cmd = $VIMRUNTIME . '\diff'
  endif
  silent execute '!' . cmd . ' ' . opt . arg1 . ' ' . arg2 . ' > ' . arg3 . eq
endfunction

let g:android_sdk_path = "C:\Program Files (x86)\Android\android-sdk"
let g:gradle_path = "D:\Programs\gradle-2.2.1\bin"
```
I've installed vim-android via Vundle `:PluginInstall`.

I open *.java file of the project using `:E` command in console vim (vim.exe).
I see messages that project is loading, or something like this.
And then I see a bunch of errors:
```
Preloading jar list cache
Error detected while processing function android#setClassPath..gradle#setClassPath..gradle#getJarList..<SNR>31_addJar..<SNR>
31_getFromCache..<SNR>31_getCache..<SNR>31_preloadCache..<SNR>31_loadCache:
line    6:
E118: Too many arguments for function: globpath
E15: Invalid expression: globpath(g:android_sdk_path . "/extras," . s:gradleCacheDir(), "**/*.jar", 1,1)
line    7:
E121: Undefined variable: l:jars
E15: Invalid expression: l:jars
line   12:
E118: Too many arguments for function: globpath
E15: Invalid expression: globpath(gradle#findRoot() . "/build/intermediates/exploded-aar", "**/classes.jar", 1, 1)
line   13:
E121: Undefined variable: l:aars
E15: Invalid expression: l:aars
Error detected while processing function android#setClassPath..gradle#setClassPath..<SNR>31_getLibJars:
line    1:
E118: Too many arguments for function: globpath
E15: Invalid expression: globpath(gradle#findRoot() . "/libs", "**/*.jar", 1,1)
Error detected while processing function android#setClassPath..gradle#setClassPath:
line   39:
E117: Unknown function: uniq
E15: Invalid expression: uniq(sort(l:jarList))
line   40:
E117: Unknown function: uniq
E15: Invalid expression: uniq(sort(l:srcList))
```

Basically plugin works: I can execute `:Android clean` and so on.
But looks like not everything works as it should.
Do you have any idea what is the cause of the problem and point me some direction?
Can you please try to open some file of the given project, just to know: is it is my own issue or common.

Thanks

**Comment by hsanson** [comment]
These are issues with your version of vim. I use the four argument globpath() and built in uniq() functions that are only available on newer versions of vim and neovim. See references for details:

  - globpath too many arguments: https://github.com/artur-shaik/vim-javacomplete2/issues/20
  - uniq method missing: https://github.com/LaTeX-Box-Team/LaTeX-Box/issues/190

Can you please test the fix-24 branch and check if it fixes your issue?

**Comment by hsanson** [comment]
Should be fixed by PR #25 .

**Comment by krossovochkin** [comment]
Hi @hsanson 
Thank you for your response and fix.
Yes, problem is now resolved.

I'll not post other Windows issues (if I find any) :)

**Comment by hsanson** [comment]
@krossovochkin good to hear it solved your issue.  And please do report issues you find. Even if I cannot fix them, I may be able to do so someday or someone else may be able to fix it.

**Comment by krossovochkin** [comment]
@hsanson OK :)
